packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -140,6 +141,18 @@ export class PersonalizeMiddleware {
   private handler = async (req: NextRequest, res?: NextResponse): Promise<NextResponse> => {
     const pathname = req.nextUrl.pathname;
     const language = req.nextUrl.locale || req.nextUrl.defaultLocale || 'en';
+
+    let pointOfSale = '';
+    try {
+      pointOfSale = PointOfSaleHelper.shouldHandleMultiValuePos(this.config.cdpConfig.pointOfSale)
+        ? PointOfSaleHelper.parseMultiValuePointOfSale(this.config.cdpConfig.pointOfSale)[language]
+        : this.config.cdpConfig.pointOfSale;
+    } catch (error) {
+      console.log(
+        'Middleware failed when parsing multi-value point of sale entry. Please check the format is correct.'
```

Comment:
You have first place one place where you log error here: https://github.com/Sitecore/jss/pull/1164/files#diff-e17fc9250ecc5aa664bb95e8348f423e640ae3e61400d39876779a4685233c61R62
and second place is here, but basically it does the same